lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -70,21 +70,22 @@ class FullPageScreenshot extends FRGatherer {
 
   /**
    * @param {LH.Gatherer.FRTransitionalContext} context
+   * @param {{height: number, width: number, mobile: boolean}} deviceMetrics
    * @return {Promise<LH.Artifacts.FullPageScreenshot['screenshot']>}
    */
-  async _takeScreenshot(context) {
+  async _takeScreenshot(context, deviceMetrics) {
     const session = context.driver.defaultSession;
     const maxTextureSize = await this.getMaxTextureSize(context);
     const metrics = await session.sendCommand('Page.getLayoutMetrics');
 
-    // Width should match emulated width, without considering content overhang.
-    // Both layoutViewport and visualViewport capture this. visualViewport accounts
-    // for page zoom/scale, which we currently don't account for (or expect). So we use layoutViewport.width.
-    // Note: If the page is zoomed, many assumptions fail.
-    //
-    // Height should be as tall as the content. So we use contentSize.height
-    const width = Math.min(metrics.layoutViewport.clientWidth, maxTextureSize);
-    const height = Math.min(metrics.contentSize.height, maxTextureSize);
+    // Height should be as tall as the content.
+    // Scale the emulated height to reach the content height.
+    const fullHeight = Math.round(
+      deviceMetrics.height *
```

Comment:
i definitely don't follow how this results in a meaningful number.... but.... i can confirm that it does indeed result in something good. 🤔

Reply:
`metrics.contentSize.height` represents the desired `metrics.layoutViewport.clientHeight` for the FPS, and we control `metrics.layoutViewport.clientHeight` by setting the emulated height over CDP.

However, the emulated height (`deviceMetrics.height`) is not necessarily equivalent to `metrics.layoutViewport.clientHeight` (because of DPR maybe).

So we use the ratio of `metrics.contentSize.height/metrics.layoutViewport.clientHeight` to determine how much we need to scale the emulated height to arrive at the desired `metrics.layoutViewport.clientHeight`.